quill-core/src/main/scala/io/getquill/norm/ExpandReturning.scala

```diff
@@ -21,7 +16,7 @@ object ExpandReturning {
     val initialExpand = ExpandReturning.apply(returning)(idiom, naming)
 
     idiom.idiomReturningCapability match {
-      case ReturningClauseSupported =>
+      case ReturningClauseSupported | OutputClauseSupported =>
```

Comment:
Make a separate clause for `OutputClauseSupported`.
Then do this:
```scala
      case  OutputClauseSupported => {
        val oldAlias = returning.alias
        val newAsts =
          initialExpand.map { case (ast, _) =>
            BetaReduction(ast, oldAlias -> Ident("INSERTED"))
          }
        implicit val namingStrategy: NamingStrategy = naming
        newAsts.map(v => idiom.translate(v))
      }
```
This will replace the `alias` in  `returning(alias => ...)` with `Ident("INSERTED")` and re-create the insert statement with the naming strategy correctly.